stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_adc.h

```diff
@@ -6259,19 +6263,43 @@ __STATIC_INLINE uint32_t LL_ADC_GetOverSamplingDiscont(ADC_TypeDef *ADCx)
   */
 __STATIC_INLINE void LL_ADC_ConfigOverSamplingRatioShift(ADC_TypeDef *ADCx, uint32_t Ratio, uint32_t Shift)
 {
-  MODIFY_REG(ADCx->CFGR2, (ADC_CFGR2_OVSS | ADC_CFGR2_OVSR), (Shift | (((Ratio - 1UL) << ADC_CFGR2_OVSR_Pos))));
+#if defined(ADC_VER_V5_V90)
+    if (ADCx == ADC3)
+    {
+        /* ADC channels ratio shift oversampling */
+        MODIFY_REG(ADCx->CFGR2, (ADC_CFGR2_OVSS | ADC3_CFGR2_OVSR), (Shift | ((Ratio >> ADC3_CFGR2_OVSR_Pos))));
```

Comment:
I'm not sure if this correct. E.g. if a ratio of 256 is required then the value written to `OVSR` should be `0b111`, and for 128 it should be `0b110`. A simple bitshift/divide will not suffice, since the relationship exponential with `2^(n+1).

Also is the bitshift supposed to move the value to the correct position? Then why `>>` and not `<<`?

However, I'm AFK at the moment so I cannot verify the code.

Reply:
'Ratio' is given from the sequence->oversampling, with possible value of 0, 1, 2,  ... 8, meaning a shift of  2^ sequence->oversampling

For the stm32H7, the ADC 3 of ADC_VER_V5_V90 : (((Ratio -1UL) << ADC3_CFGR2_OVSR_Pos))
Ratio = sequence->oversampling = 8 (oversampling 256x), the OVSR[2:0] =  0b111
Ratio = sequence->oversampling = 6 (oversampling 64x), the OVSR[2:0] =   0b101
Ratio = sequence->oversampling = 4 (oversampling 16x), the OVSR[2:0] =   0b011

for other ADC : (((1UL << Ratio) - 1UL) << ADC_CFGR2_OVSR_Pos
Ratio = sequence->oversampling = 8 (oversampling 256x), the OSVR[9:0] =  255
Ratio = sequence->oversampling = 6 (oversampling 64x), the OSVR[9:0] =  63
Ratio = sequence->oversampling = 4 (oversampling 16x), the OSVR[9:0] =  15